[READY] Update onboarding docs to improve reader experience (#2396)
kedro-org/kedro: docs/source/get_started/install.md

## Create a Python virtual environment for your Kedro project

We strongly recommend [installing `conda` as your virtual environment manager](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) if you don't already use it.

``` {tip}
[Read more about Python virtual environments](https://realpython.com/python-virtual-environments-a-primer/) or [watch an explainer video about them](https://youtu.be/YKfAwIItO7M).
```

To create a new virtual environment called `kedro-environment` using `conda`:

```bash
conda create --name kedro-environment python=3.10 -y
```

In this example, we use Python 3.10, but you can opt for a different version if you need it for your particular project.

The `conda` virtual environment is not dependent on your current working directory and can be activated from any directory:

```bash
conda activate kedro-environment
```

To confirm that a valid version of Python is installed in your virtual environment, type the following in your terminal (macOS and Linux):

```bash
python3 --version
```

On Windows:

```bash
python --version
```

To exit `kedro-environment`:

```bash
conda deactivate
```

> ### Create a new Python virtual environment without using `conda`

Author: @astrojuanlu Please can you review this section?

* **Python virtual environment**: We suggest you create a new Python virtual environment for *each* new Kedro project you work on to isolate its dependencies from those of other projects.

* **git**: You must install `git` onto your machine if you do not already have it. Type `git -v` into your terminal window to confirm it is installed. You should see the version of `git` available or an error message. You can download it from the official  [`git`](https://git-scm.com/) website.

## Create a Python virtual environment for your Kedro project

We strongly recommend [installing `conda` as your virtual environment manager](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) if you don't already use it.

``` {tip}
[Read more about Python virtual environments](https://realpython.com/python-virtual-environments-a-primer/) or [watch an explainer video about them](https://youtu.be/YKfAwIItO7M).
```

### How to create a new virtual environment using `conda`

This is our recommended approach. From your terminal:

```bash
conda create --name kedro-environment python=3.10 -y
```

> In this example, we use Python 3.10, and create a virtual environment called `kedro-environment`. You can opt for a different version of Python if you need it for your particular project, and name it anything you choose.

Author: ```suggestion
In this example, we use Python 3.10, and create a virtual environment called `kedro-environment`. You can opt for a different version of Python (any version >= 3.7 and <3.11) for your project, and name it anything you choose.
```

> In this example, we use Python 3.10, and create a virtual environment called `kedro-environment`. You can opt for a different version of Python (any version >= 3.7 and <3.11) for your project, and name it anything you choose.

Reviewer: I get that _we_ are setting some recommendations here (rather than enforcing readers to do a specific thing) but I'm having a hard time reading all the first-person sentences https://ddbeck.com/first-person-plural-is-questionable/

Wondering if we could rephrase to reduce those, what do you think?

Author: Agreed! I'll do a final grammatical run through.

The `conda` virtual environment is not dependent on your current working directory and can be activated from any directory:

```bash
conda activate kedro-environment
```

To confirm that a valid version of Python is installed in your virtual environment, type the following in your terminal (macOS and Linux):

```bash
python3 --version
```

On Windows:

```bash
python --version
```

To exit `kedro-environment`:

```bash
conda deactivate
```

> ### How to create a new Python virtual environment without using `conda`

Reviewer: This is actually OK, becuse `venv` and Pipenv are Python-specific

Depending on your preferred Python installation, you can create virtual environments to work with Kedro using `venv` or `pipenv` instead of `conda`.

<details>
<summary><b>Click to expand instructions for <code>venv</code></b></summary>

If you use Python 3, you should already have the `venv` module installed with the standard library. Create a directory for working with your project and navigate to it. For example:

```bash
mkdir kedro-environment && cd kedro-environment
```

Next, create a new virtual environment in this directory with `venv`:

> ```bash
> python -m venv env/kedro-environment  # macOS / Linux
> python -m venv env\kedro-environment  # Windows

Reviewer: Let's avoid the extra nesting here:

````suggestion
```bash
python -m venv .venv
```
````

(and also this suggestion is cross platform)

```

Activate this virtual environment:

> ```bash
> source env/kedro-environment/bin/activate # macOS / Linux
> .\env\kedro-environment\Scripts\activate  # Windows
> ```

Reviewer: Consistent with the above:

````suggestion
```bash
source .venv/bin/activate # macOS / Linux
.\.venv\Scripts\activate  # Windows
```
````

Reviewer: This is a "weird" workflow still because the `.venv` will live outside the project directory itself after `kedro new` is called. But users will need to cope with this unfortunately.

Should we add a note about that?
